src/file_dialog.rs

```diff
@@ -376,14 +376,24 @@ impl FileDialog {
             .id
             .map_or_else(|| egui::Id::new(self.get_window_title()), |id| id);
 
-        self.load_directory(&self.get_initial_directory());
+        let init_dir = self.get_initial_directory();
+        self.load_directory(&init_dir);
+        let init_path = &self.config.initial_path;
+        // If the initial path has an extra component, use it to pre-select an item.
```

Comment:
I'm honestly not sure if this is the best way to go about it, or if we should do a more clear separation of initial directory, and initially selected file at a higher level.